### PR TITLE
refactor(java): Detach MemoryBuffer from Fury

### PR DESCRIPTION
**Pull request by LiangliangSui**
## What does this PR do?

Separate MemoryBuffer management tasks from Fury and reduce Fury class responsibilities.

## Does this PR introduce any user-facing change?

- [ ] Does this PR introduce any public API change?
- [ ] Does this PR introduce any binary protocol compatibility change?

### CODE REVIEW

File: java/fury-core/src/main/java/org/apache/fury/Fury.java [hunk 1]
```diff
@@ -117,6 +116,7 @@ public final class Fury implements BaseFury {
   private Iterator<MemoryBuffer> outOfBandBuffers;
   private boolean peerOutOfBandEnabled;
   private int depth;
+  private final MemoryBufferManager bufferManager;
```
Comment (chaokunyang) [inline java/fury-core/src/main/java/org/apache/fury/Fury.java:119]: The buffer logic is pretty simple, I don't think it deserves a separate abstraction.  The abstraction is not always a good thing.

Comment (LiangliangSui) [reply java/fury-core/src/main/java/org/apache/fury/Fury.java:119]: Could we apply `execute` method in `Fury`?

Comment (chaokunyang) [reply java/fury-core/src/main/java/org/apache/fury/Fury.java:119]: I prefer not, we should reduce callstack and avoid use lambda here to reduce performance cost. There are alreay several nested call before the real serialization logic run

Comment (LiangliangSui) [reply java/fury-core/src/main/java/org/apache/fury/Fury.java:119]: Okay, I got it, I will close this PR later.

**Comment by LiangliangSui** [comment]
What can I do for Fury currently? Implement the design in [xlang_serialization_spec.md](https://github.com/apache/incubator-fury/blob/main/docs/specification/xlang_serialization_spec.md)?

**Comment by chaokunyang** [comment]
> What can I do for Fury currently? Implement the design in [xlang_serialization_spec.md](https://github.com/apache/incubator-fury/blob/main/docs/specification/xlang_serialization_spec.md)?

Yep, we basically redesigned our xlang serialization protocol, the implementation in java/python/golang/javascript are not alighed to the new proposal. Maybe we can take over some features development in the new protocol

**Comment by LiangliangSui** [comment]
> > What can I do for Fury currently? Implement the design in [xlang_serialization_spec.md](https://github.com/apache/incubator-fury/blob/main/docs/specification/xlang_serialization_spec.md)?
> 
> Yep, we basically redesigned our xlang serialization protocol, the implementation in java/python/golang/javascript are not alighed to the new proposal. Maybe we can take over some features development in the new protocol

Okay, I'll take some of the features and implement it.

**Pull request closed without merging**